test_conformance/extensions/cl_khr_command_buffer/negative_command_buffer_fill.cpp

```diff
@@ -0,0 +1,461 @@
+                                      fill_color_1, origin, region, 0, nullptr,
+                                      nullptr, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_CONTEXT,
+                               "clCommandFillImageKHR should return "
+                               "CL_INVALID_CONTEXT",
+                               TEST_FAIL);
+
+        command_buffer = clCreateCommandBufferKHR(1, &queue1, 0, &error);
+        test_error(error, "clCreateCommandBufferKHR failed");
+
+        error = clCommandFillBufferKHR(command_buffer, nullptr, out_mem,
+                                       &pattern, sizeof(cl_int), 0, data_size(),
+                                       0, nullptr, nullptr, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_CONTEXT,
+                               "clCommandFillBufferKHR should return "
+                               "CL_INVALID_CONTEXT",
+                               TEST_FAIL);
+
+        error = clCommandFillImageKHR(command_buffer, nullptr, dst_image,
+                                      fill_color_1, origin, region, 0, nullptr,
+                                      nullptr, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_CONTEXT,
+                               "clCommandFillImageKHR should return "
+                               "CL_INVALID_CONTEXT",
+                               TEST_FAIL);
+
+        return CL_SUCCESS;
+    }
+
+    cl_int SetUp(int elements) override
+    {
+        cl_int error = BasicCommandBufferTest::SetUp(elements);
+        test_error(error, "BasicCommandBufferTest::SetUp failed");
+
+        context1 = clCreateContext(0, 1, &device, nullptr, nullptr, &error);
+        test_error(error, "Failed to create context");
+
+        out_mem_ctx =
+            clCreateBuffer(context1, CL_MEM_WRITE_ONLY,
+                           sizeof(cl_int) * num_elements, nullptr, &error);
+        test_error(error, "clCreateBuffer failed");
+
+        dst_image_ctx = create_image_2d(context1, CL_MEM_WRITE_ONLY, &formats,
+                                        512, 512, 0, NULL, &error);
+        test_error(error, "create_image_2d failed");
+
+        queue1 = clCreateCommandQueue(context1, device, 0, &error);
+        test_error(error, "clCreateCommandQueue failed");
+
+        dst_image = create_image_2d(context, CL_MEM_WRITE_ONLY, &formats, 512,
+                                    512, 0, NULL, &error);
+        test_error(error, "create_image_2d failed");
+
+        return CL_SUCCESS;
+    }
+
+    clContextWrapper context1;
+    clCommandQueueWrapper queue1;
+    clMemWrapper out_mem_ctx;
+    clMemWrapper dst_image;
+    clMemWrapper dst_image_ctx;
+    const cl_uint fill_color_1[4] = { 0x05, 0x05, 0x05, 0x05 };
+    const cl_image_format formats = { CL_RGBA, CL_UNSIGNED_INT8 };
+    const size_t origin[3] = { 0, 0, 0 };
+    const size_t region[3] = { 512, 512, 1 };
+    const cl_int pattern = 0xFF;
```

Comment:
If these are identical between all the classes can we define them in a single place, such as a base class that is inherited from.